src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1331,9 +1331,14 @@ static struct comp_dev *volume_new(const struct comp_driver *drv,
 static void volume_legacy_free(struct comp_dev *dev)
 {
 	struct processing_module *mod = comp_get_drvdata(dev);
+	struct module_data *md = &mod->priv;
+	struct module_config *cfg = &md->cfg;
 
 	comp_dbg(dev, "volume_legacy_free()");
 
+	if (cfg->data)
+		rfree(cfg->data);
```

Comment:
@ranj063 this now checks `if (cfg->data)` twice... Would be good to do a follow-up to remove this check

Reply:
Can we make sure Zephyr does the same check for ```free()``` too (and then we are safe).